src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -709,51 +702,65 @@ protected function processRefunds($woocommerceOrderId, $storekeeperId): void
                 }
             }
         }
 
         $unsyncedRefundsWithoutPaymentIds = PaymentGateway::getUnsyncedRefundsWithoutPaymentIds($woocommerceOrderId);
         if (count($unsyncedRefundsWithoutPaymentIds) > 0) {
             foreach ($unsyncedRefundsWithoutPaymentIds as $unsyncedRefundsWithoutPaymentId) {
-                $payOrderRefundId = false;
                 try {
                     $id = $unsyncedRefundsWithoutPaymentId['id'];
                     $refundId = $unsyncedRefundsWithoutPaymentId['wc_refund_id'];
                     $refundAmount = $unsyncedRefundsWithoutPaymentId['amount'];
                     if (PaymentGateway::hasPayment($woocommerceOrderId)) {
                         $storekeeperPaymentId = PaymentGateway::getPaymentId($woocommerceOrderId);
-                        $storekeeperRefundId = $shopModule->refundAllOrderItems([
-                            'id' => $storekeeperId,
-                            'refund_payments' => [
-                                [
-                                    'payment_id' => $storekeeperPaymentId,
-                                    'amount' => -abs($refundAmount),
-                                    'description' => sprintf(
-                                        __('Refund via Wordpress plugin (Refund #%s)', I18N::DOMAIN),
-                                        $refundId
-                                    ),
+                        try {
+                            $storekeeperRefundId = $shopModule->refundAllOrderItems([
+                                'id' => $storekeeperId,
+                                'refund_payments' => [
+                                    [
+                                        'payment_id' => $storekeeperPaymentId,
+                                        'amount' => round(-abs($refundAmount), 2),
+                                        'description' => sprintf(
+                                            __('Refund via Wordpress plugin (Refund #%s)', I18N::DOMAIN),
+                                            $refundId
+                                        ),
+                                    ],
                                 ],
-                            ],
-                        ]);
-                        $payOrderRefundId = PaymentGateway::updateRefund($id, $storekeeperRefundId, $refundAmount);
+                            ]);
+                            PaymentGateway::updateRefund($id, $storekeeperRefundId, $refundAmount);
+                        } catch (GeneralException $generalException) {
+                            if ('Only invoiced orders can be refunded' === $generalException->getMessage()) {
+                                $storekeeperRefundId = PaymentGateway::createRefundAsPayment($refundId, $refundAmount);
+                                PaymentGateway::updateRefund($id, $storekeeperRefundId, $refundAmount);
+
+                                try {
```

Comment:
do not do nested try catching (here you have 3rd level :P), usually when you need it, it's a good indicator that you should put it inside a function. especially that you reuse the `$generalException` between those

Reply:
Moved to a function and removed nested try catches.